src/llvm/basic_block_collection.cr

```diff
@@ -16,4 +16,27 @@ struct LLVM::BasicBlockCollection
     yield builder
     block
   end
+
+  def each : Nil
+    bb = LibLLVM.get_first_basic_block(@function)
+    while bb
+      yield LLVM::BasicBlock.new bb
+      bb = LibLLVM.get_next_basic_block(bb)
+    end
+  end
+
+  def []?(name : String)
+    self.each do |bb|
+      return bb if bb.name == name
+    end
```

Comment:
After including `Enumerable` module you could shorten it to: `find(&.name.==(name))`.

Reply:
Good to know! Thank you!